### ISSUE TITLE
the attention alignment do not converge while outputs_per_step=1

### ISSUE THREAD

**Issue by Jim-Song** [Rayhane-mamah/Tacotron-2#175]
the attention alignment converge in 35000 steps, while r = outputs_per_step = 2, like this:
![image](https://user-images.githubusercontent.com/32925029/44637627-e5edba80-a9e4-11e8-9ec6-5d66d1a7a453.png)
and converge to a very good rusult after 105000 steps, like this:
  
![image](https://user-images.githubusercontent.com/32925029/44637666-1b92a380-a9e5-11e8-9433-a97c36724c52.png)

but while r = 1, the alignment do not converge after 70000 steps in two separate experiments, like this:

 
![image](https://user-images.githubusercontent.com/32925029/44637718-6a403d80-a9e5-11e8-82a7-0fd55a82fe7b.png)

![image](https://user-images.githubusercontent.com/32925029/44637729-762bff80-a9e5-11e8-8e7d-d63ceebd6265.png)

in the paper, r = 1. Dose any one has a good result while r = 1?

by the way, I used keithito's tacotron1 and the alignment do not converge while r = 2, only converge while r = 3, 4, 5

**Comment by begeekmyfriend** [comment]
Did you shrink the batch size? In my experience, the tacotron batch size might not be less than 32.

**Comment by Jim-Song** [comment]
@begeekmyfriend hi, I used the **ljspeech** corpus to train the tacotron model and the **batch_size is 40**. I began a new training with the tacotron_initial_**learning_rate**=1e-4,tacotron_**final_learning_rate**=1e-6, but there is no sign of convergence. (Totally used the latest commit )

![image](https://user-images.githubusercontent.com/32925029/44760763-ec11a180-ab72-11e8-9b83-330fd6116a40.png)

**Comment by begeekmyfriend** [comment]
@Jim-Song Please wait until 40K steps. Or you might set `symmetric_mels = True` in `hparam.py`. I am using `r=2` and `batch size = 32`.
![step-7000-align 2](https://user-images.githubusercontent.com/6031938/44760980-e9fc1280-ab73-11e8-9f50-e0afb8c42fe2.png)

**Comment by Jim-Song** [comment]
@begeekmyfriend  thx, I will try it now

**Comment by Jim-Song** [comment]
@begeekmyfriend the r=1 case alignment still do not converge after 45000 steps. 

I noticed that in the tacotron2 paper, the linear loss was not included, but in the latest commit, the loss includes a loss of linear_outputs from mel_outputs by a cbhg. In your experiment ,was the linear loss included?

**Comment by begeekmyfriend** [comment]
Of course the linear loss included. Then try `symmetric_mels = True`

**Comment by Jim-Song** [comment]
@begeekmyfriend after 8 days' training, it finally converge
![image](https://user-images.githubusercontent.com/32925029/45203702-8a0e1600-b2af-11e8-990e-2eb528efa59b.png)
and the generated video samples are also better

**Comment by begeekmyfriend** [comment]
Which step did it achieve first alignment?

**Comment by Jim-Song** [comment]
![image](https://user-images.githubusercontent.com/32925029/45272478-f2433e80-b4df-11e8-842d-94a357112b43.png)
![image](https://user-images.githubusercontent.com/32925029/45272485-fff8c400-b4df-11e8-8811-858dcd57043d.png)
140000-180000steps
@begeekmyfriend

**Comment by wuzhonglijz** [comment]
@begeekmyfriend Why you can be close to ~250 of Y-axis？Mine and others just 100~160.

**Comment by begeekmyfriend** [comment]
@wuzhonglijz It depend on the length of the training wav clips.

**Comment by Mouhamed-Hamed** [comment]
@Jim-Song I am new to Tacotron2, Please I want to know where I can found r=1 ?

**Comment by debasishaimonk** [comment]
which attention did u use @Jim-Song ? is it Graves attention or Bahadanu attention?
